src/octoprint/util/comm.py

```diff
@@ -3240,10 +3274,55 @@ def _send_loop(self):
 					# are done processing the last fetched queue entry
 					self._send_queue.task_done()
 
-				# now we just wait for the next clear and then start again
-				self._clear_to_send.wait()
+					if ( self._AdvancedOkSupported ):
```

Comment:
This whole section needs comments because otherwise in 6 months whoever ends up maintaining this (probably me) will have *no* idea anymore what is going on here and why the conditions for the `if`s are defined as they are.